File: src/main/java/net/dontdrinkandroot/gitki/model/LockInfo.java
package net.dontdrinkandroot.gitki.model;

import java.time.Instant;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class LockInfo
{
    private FilePath path;

    private GitUser user;

    private Instant expiry;

    public LockInfo(FilePath path, GitUser user, Instant expiry)
    {
        this.path = path;
        this.user = user;
        this.expiry = expiry;
    }

    public FilePath getPath()
    {
        return this.path;
    }

    public GitUser getUser()
    {
        return this.user;
    }

    public Instant getExpiry()
    {
        return this.expiry;
    }

    public void setExpiry(Instant expiry)
    {
        this.expiry = expiry;
    }

    public boolean isExpired()
    {
        return this.expiry.isBefore(Instant.now());
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/model/LockInfo.kt
package net.dontdrinkandroot.gitki.model

import java.io.Serializable
import java.time.Instant

class LockInfo(val path: FilePath, val user: GitUser, var expiry: Instant) : Serializable {

    val expired: Boolean
        get() = expiry.isBefore(Instant.now())
}